### DISCUSSION TITLE
Use YSU PBL scheme in UFS SRW.

### DISCUSSION THREAD

**Discussion by lo-y-wni** [ufs-community/ufs-srweather-app#723] [Q&A] [answered]
Hello all developers and users:

I want to ask how can I use YSU PBL scheme in UFS SRW? I have checked the code and suites in CCPP directory and I confirmed YSU is available in CCPP but after I read the science document, there is no any information about how to use YSU scheme.

I got an example of suite in ccpp/suites/suite_FV3_GFS_2017_ysu.xml but unfortunately there is no namelist example. How do I set the namelist in FV3 to use YSU?

Yi-Chuan

**Comment by gspetro-NOAA** [comment]
Hi Yi-Chuan, 

Currently, the YSU PBL scheme is not available in the SRW App, although it is available in the Weather Model. However, it should be possible to add it. I've reached out to @EdwardSnyder-NOAA, and he is looking into this for you. In the meantime, you may find some helpful information in [these talks by the DTC](https://dtcenter.org/events/2021/unified-forecast-system-ufs-short-range-weather-srw-application-users-training/agenda), specifically the CCPP ones by Linlin Pan and Laurie Carson (Overview of the CCPP and CCPP for Developers, respectively). The talks are for an older version of the SRW App, but they may still contain some helpful information while you wait for Eddie to get back to you. 

Best,
Gillian Petro, Advanced User Support (AUS) Team

**Comment by EdwardSnyder-NOAA** [comment]
Good morning Yi-Chuan,

The YSU PBL scheme can be enabled with the following SRW App supported physics schemes: `FV3_GFS_v15p2`, `FV3_GFS_v16`, or `FV3_GFS_v17_p8`. 

To enable the YSU PBL scheme, go to `ufs-srweather-app/parm/FV3.input.yml` and set the `do_ysu` variable to `True` under the physics scheme you would like to use. Ensure you are using the same physics scheme in your `config.yaml` file. Then run the `generate_FV3LAM_wflow.py` script. Once in the experiment directory, view the namelist file and you should see `do_ysu = .true.`, which indicates that the YSU PBL scheme is enabled. 

Please let us know if you have any additional questions.

Thanks again!

Eddie | Advanced User Support (AUS) Team

**Reply by lo-y-wni** [reply]
Good morning Eddie: 

Thanks a lot! It looks it's very easy to use YSU scheme, but before setting do_ysu to True, should I modify the SDFs of  `FV3_GFS_v15p2`, `FV3_GFS_v16` and `FV3_GFS_v17_p8`? Taking `FV3_GFS_v16` for example, the original PBL scheme is TKE-EDMF, but to use YSU, is it necessary to replace `<scheme>satmedmfvdifq</scheme>` with `<scheme>ysuvdif</scheme>`?

Yi-Chuan

**Reply by EdwardSnyder-NOAA** [reply]
You shouldn't need to modify the SDFs files. The SRW App uses a `set_namelist.py` script which generates the experiment's namelist file from a base namelist and the `FV3.input.yml` file. This yaml contains the information from the supported physics scheme's SDFs plus a number of configurable variables. 

In regard to your second question, you would need to disable the satmedmfvdifq variable, which can be done by going into `ufs-srweather-app/parm/FV3.input.yml` file and turning the `satmedmf` variable to `False` under the physics scheme you are using. 

Please let us know if you have any additional questions!

Thanks!

Eddie | Advanced User Support (AUS) Team

**Reply by lo-y-wni** [reply]
I dig into the fortran code of GFS_v16 (ccpp_FV3_GFS_v16_physics_cap.F90) but I don't see there is any subroutine calling ysu scheme but calling satmedmf scheme. If I just set satmedmf to False, it seems it won't call any PBL scheme any more. However, if I set 
`<scheme>ysuvdif</scheme>` and re-compile the code, I can see `ysuvdif_run` being called in the code.

Yi-Chuan

**Reply by EdwardSnyder-NOAA** [reply, marked as answer]
My apologies Yi-Chuan. I now see that too. I'd recommend updating the SDF file and re-compiling the code.

Thanks for find that!

Eddie Snyder | Advanced User Support (AUS) Team
